### PR TITLE
Add transformer basic example

### PR DESCRIPTION
**Pull request by graemenail**
Adds a basic transformer example using SentencePiece. 

The model uses the transformer-base task as a starting point, and adjusts the configuration to validate and save more frequently for demonstration. It uses a subset of the WMT21 en-de news-task data, with WMT19 as a validation set and WMT20 as a test set. Data cleaning is minimal, with non-printing characters removed, length constraint of 1-100 words, and deduplication. The data consists of EuroParl, news-commentary and commoncrawl.

It trains for 10 epoch, or 10 stalled validations on ce-mean-word, whichever occurs first. It validates on ce-mean-word, bleu and perplexity. This took around 8hr on 2 RTX 2080s with a workspace of 9000MiB. 

The results of the model are shown for WMT16-20, for BLEU, chrF2 and Comet (`wmt20-comet-da`).

There is a comet-script provided for convenience as well. It is written such that it can also be used as a validation script for Marian: i.e. `./comet-score /tmp/marian-outut.txt` is scored against the validation set. I found that when forced to execute on the CPU, it was a little slow for this example.

### CODE REVIEW

File: transformer-intro/README.md [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+
+## Getting Marian
+The development version of Marian can be obtained with
+```shell
+git clone https://github.com/marian-nmt/marian-dev
+cd marian-dev
+```
+
+### Compile
+To compile Marian we need to ensure we have the required packages. The list of requirements can be found in the [documentation][install_marian]. Since we're using SentencePiece, we also need to make sure we have satisfy its[requirements][install_sentencepiece] too.
```
Comment (snukky) [inline transformer-intro/README.md]:
```suggestion
To compile Marian we need to ensure we have the required packages. The list of requirements can be found in the [documentation][install_marian]. Since we're using SentencePiece, we also need to make sure we have satisfy its [requirements][install_sentencepiece] too.
```

File: transformer-intro/README.md [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+In this example we will use Marian to create a English-German translation
+system. We'll follow a very simple pipeline with data acquisition, some basic
+corpus cleaning, generation of vocabulary with [SentencePiece], training of a
+transformer model, and evaluation with [sacreBLEU], and (optionally) [Comet].
+
+We'll be using a subset of data from the WMT21 [news task] to train our model.
+For the validation and test sets, we'll use the test sets from WMT19 and WMT20,
+respectively.
+
+Lets get started by installing our dependencies!
```
Comment (snukky) [inline transformer-intro/README.md:15]: If it aims for beginners, you may add that this tutorial has been prepared to be run on Linux and a Nvidia GPU is required.

File: transformer-intro/README.md [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+
+Then we can compile with
+```shell
+mkdir build
+cd build
+cmake .. -DCMAKE_BUILD_TYPE=Release -DUSE_SENTENCEPIECE=ON
+cmake --build .
+```
+
+To speed up compilation we can use ```cmake --build . -j 8``` to run 8 tasks simultaneously.
```
Comment (snukky) [inline transformer-intro/README.md]: Consider mentioning that one may want to decrease the number of tasks depending on the machine.

File: transformer-intro/README.md [hunk 4]
```diff
@@ -0,0 +1,389 @@
+# Basic Transformer
+
+In this example we will use Marian to create a English-German translation
```
Comment (snukky) [inline transformer-intro/README.md:6]: Consider mentioning that this is a basic tutorial that should be a good fit for new practitioners into Marian and machine translation in general.

**Review by snukky — APPROVED** [review]

**Comment by graemenail** [comment]
Thanks for the comments @snukky ; I've addressed them all.

Merging
